### PR TITLE
expression: open new expression framework by default

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
LGTM

File: expression/builtin.go [hunk 1]
```diff
@@ -221,76 +219,83 @@ func (b *baseBuiltinFunc) getArgs() []Expression {
...
+	case tpDecimal:
+		res, isNull, err = b.self.evalDecimal(row)
+	case tpDatetime, tpTimestamp:
+		res, isNull, err = b.self.evalTime(row)
+	case tpDuration:
+		res, isNull, err = b.self.evalDuration(row)
+	case tpJSON:
+		res, isNull, err = b.self.evalJSON(row)
+	case tpString:
+		res, isNull, err = b.self.evalString(row)
```
Comment (winkyao) [inline expression/builtin.go]: default?

Comment (zz-jason) [reply expression/builtin.go]: Yes, mainly copied from `ScalarFunction.Eval()`. This function now only used in unit tests, we should remove this API in the future.

**Pull request merged**